### DISCUSSION TITLE
Converters for Non-Supported Frameworks

### DISCUSSION THREAD

**Discussion by johanndm1** [onnx/onnx#5255] [Q&A]
I have a model built in a framework currently unsupported by ONNX (ReservoirPy). Is there a process for creating the converter library for unsupported frameworks? If not, how may I implement my model in ONNX?

**Comment by justinchuby** [comment]
I recommend looking at the ONNX Script project https://github.com/microsoft/onnxscript for ways to express the operators in the framework you use in ONNX. Assuming you are working with a computational graph, the basic idea would be going through all nodes in the graph and translate each node to its ONNX equivalent; then fill in the rest of the info (weights, inputs etc) to complete the ONNx graph.

**Reply by justinchuby** [reply]
It’d be interesting to see reservoir models expressed in ONNX! For any iterative processes in the models, a Loop node or something external to the ONNx model that drives the iteration may be needed.

**Comment by xadupre** [comment]
I'm not aware of any converting library for reservoirpy. You may need to write one for this library or you could translate your models into another model based on scikit-learn or torch, both have a dedicated converting library (torch.onnx, sklearn-onnx).
